src/System.Windows.Forms/src/System/Windows/Forms/Controls/Labels/LinkLabel.cs

```diff
@@ -509,7 +510,7 @@ private static int ConvertToCharIndex(int index, string text)
     {
         Debug.Assert(g is not null);
 
-        if (_textRegion is not null)
+        if (_textLayoutValid && _textRegion is not null)
```

Comment:
It's valid to have an empty region when the linklable does not have any text. See this repro case that @Epica3055 had investigated - https://github.com/dotnet/winforms/issues/10515#issuecomment-1888344874
```suggestion
        if (_textLayoutValid)
```

Reply:
@LeafShi1 and @Epica3055  - This PR might fix both https://github.com/dotnet/winforms/issues/10515 and https://github.com/dotnet/winforms/issues/10630, please verify that and add 2 unit tests to this PR.